nemo-physical/src/resource.rs

```diff
@@ -1,10 +1,502 @@
+                        )),
+                    );
+                }
+
+                let mut idx = 0;
+                while let Some(element) = tuple.tuple_element(idx) {
+                    idx += 1;
+                    // Validate ValueDomain of each element in tuple
+                    if !matches!(
+                        element.value_domain(),
+                        ValueDomain::PlainString | ValueDomain::Int | ValueDomain::Iri
+                    ) {
+                        return Err(
+                            ResourceValidationErrorKind::ImportExportInvalidHttpParameter(format!(
+                                "element in tuple has wrong type: {:?}",
+                                element.value_domain()
+                            )),
+                        );
+                    }
+                }
+            }
+            Ok(())
+        } else {
+            // Return Ok, if the parameter was empty
+            Ok(())
+        }
+    }
+
+    /// Convert Header-Parameter into a key-value vector
+    pub fn unpack_header_parameter(map: AnyDataValue) -> impl Iterator<Item = (String, String)> {
+        // Expect a map of key-value pairs
+        let keys = map.map_keys();
+
+        let mut vec: Vec<(String, String)> = Vec::new();
+        if let Some(keys_iter) = keys {
+            for key in keys_iter {
+                let element = map.map_element_unchecked(key);
+                vec.push((key.lexical_value(), element.lexical_value()));
+            }
+        }
+        vec.into_iter()
+    }
+
+    /// Flatten each parameter-map into a key-value vector
+    pub fn unpack_get_post_parameter(map: AnyDataValue) -> impl Iterator<Item = (String, String)> {
+        // Expect each value as a tuple (of values), where each element can be converted into a string
+        let keys = map.map_keys();
+
+        let mut vec: Vec<(String, String)> = Vec::new();
+        if let Some(keys_iter) = keys {
+            for key in keys_iter {
+                let value = map.map_element_unchecked(key);
+                let mut idx = 0;
+                // Unpack inner tuple
+                while let Some(element) = value.tuple_element(idx) {
+                    idx += 1;
+                    vec.push((key.lexical_value(), element.lexical_value()));
+                }
+            }
+        }
+        vec.into_iter()
+    }
+
+    /// Add key-value pair to HeaderParameter
+    pub fn header_mut(
+        &mut self,
+        key: String,
+        value: String,
+    ) -> Result<&mut Self, ResourceValidationErrorKind> {
+        if self.headers.iter().any(|(k, _)| k == &key) {
+            Err(ResourceValidationErrorKind::ImportExportDuplicateHttpHeader(key))
+        } else {
+            self.headers.push((key, value));
+            Ok(self)
+        }
+    }
+
+    /// Add key-value pair to GetParameter
+    pub fn get_mut(&mut self, key: String, value: String) -> &Self {
```

Comment:
* `should be called `add_query_parameter` or `add_get_parameter`
* should return `&mut Self`, so that calls can be chained